gnc/sub8_perception/src/sub8_pcl/cv_utils.cc

```diff
@@ -2,499 +2,655 @@
+    }
+    hist.at<float>(i) = sum;
+  }
+  for (int i = 0; i < offset; ++i) // Pad filtered result with zeroes
+  {
+    hist.at<float>(i) = 0;
+    hist.at<float>(histSize - 1 - i) = 0;
+  }
+  return hist;
 }
 
-
-std::vector<float> generate_gaussian_kernel_1D(size_t kernel_size, float sigma){
-	std::vector<float> kernel;
-	int middle_index = (kernel_size - 1) / 2;
-	int first_discrete_sample_x = -(middle_index);
-	for (int i = first_discrete_sample_x; i <= 0; i++)
-	{
-		float power = -0.5 * (float(i)/sigma) * (float(i)/sigma);
-		kernel.push_back( exp(power) );				// From definition of Standard Normal Distribution
-	}
-	for(int i = 1; i <= middle_index; i++){			// Kernel is symmetric
-		kernel.push_back(kernel[middle_index - i]);
-	}
-	// Normalize kernel (sum of values should equal 1.0)
-	float sum = 0;
-	for (size_t i = 0; i < kernel_size; i++){ sum += kernel[i];	}
-	for (size_t i = 0; i < kernel_size; i++){ kernel[i] /= sum;	}
-	return kernel;
+std::vector<float> generate_gaussian_kernel_1D(size_t kernel_size,
+                                               float sigma) {
+  std::vector<float> kernel;
+  int middle_index = (kernel_size - 1) / 2;
+  int first_discrete_sample_x = -(middle_index);
+  for (int i = first_discrete_sample_x; i <= 0; i++) {
+    float power = -0.5 * (float(i) / sigma) * (float(i) / sigma);
+    kernel.push_back(
+        exp(power)); // From definition of Standard Normal Distribution
+  }
+  for (int i = 1; i <= middle_index; i++) { // Kernel is symmetric
+    kernel.push_back(kernel[middle_index - i]);
+  }
+  // Normalize kernel (sum of values should equal 1.0)
+  float sum = 0;
+  for (size_t i = 0; i < kernel_size; i++) {
+    sum += kernel[i];
+  }
+  for (size_t i = 0; i < kernel_size; i++) {
+    kernel[i] /= sum;
+  }
+  return kernel;
 }
 
-
-std::vector<cv::Point> find_local_maxima(const cv::MatND &histogram, float thresh_multiplier){
-	std::vector<cv::Point> local_maxima, threshed_local_maxima;
-	float global_maximum = -std::numeric_limits<double>::infinity();
-
-	// Locate local maxima and find global maximum
-	for(size_t idx = 1; idx < histogram.total() - 1; idx++){
-		float current_value = histogram.at<float>(idx);
-	    if((histogram.at<float>(idx-1) < current_value) && (histogram.at<float>(idx+1) <= current_value)){
-	    	local_maxima.push_back(cv::Point(idx, current_value));
-	    	if(global_maximum < current_value) global_maximum = current_value;
-	    }
-	}
-#ifdef SEGMENTATION_DEBUG
-	std::cout << std::endl << "Maxima: ";
-#endif
-	BOOST_FOREACH(cv::Point pt, local_maxima){
-		if(pt.y > global_maximum * thresh_multiplier) threshed_local_maxima.push_back(pt);
-#ifdef SEGMENTATION_DEBUG
-		std::cout << boost::format("[%1%, %2%] ") % pt.x % pt.y;
-#endif
-	}
+std::vector<cv::Point> find_local_maxima(const cv::MatND &histogram,
+                                         float thresh_multiplier) {
+
+  std::stringstream ros_log;
+  ros_log << "\x1b[1;31m"
```

Comment:
Correct, debugging segmentation stuff with color coding is much more pleasant and efficient. It might print out a few garbage characters when we log to files but I'm fine with that.